### DISCUSSION TITLE
No works with multiple users ID or Usernames

### DISCUSSION THREAD

**Discussion by rafageist** [spatie/twitter-streaming-api#38] [Q&A]
```
PublicStream::create(
    $bearerToken,
    $apiKey,
    $apiSecretKey
)->whenTweets(['user1','user2',...], function(array $data) ....
```

**Comment by rafageist** [comment]
The problem is the resulting rule:

If you pass ['user1','user2', ...] the resulting rule is: from:user1 from:user2 without operator, and the documentation say

Matches any Tweet from a specific user.
The value can be either the username (excluding the @ character) or the user’s numeric user ID.

You can only pass a single username/ID from: operator.

Example: from:twitterdev OR from:twitterapi -from:twitter

https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/integrate/build-a-rule

**Comment by rafageist** [comment]
Maybe you can do something like this change:

```php
public function whenTweets(string | array $twitterUserIds, callable $whenTweets): self
    {
        $i = 0;
        if (is_array($twitterUserIds)) {
            foreach ($twitterUserIds as $id) {
                $i++;
                if ($i > 1) {
                    $this->rule->or();
                }
                
                $this->rule->from($id);
            }
        }
        else {
            $this->rule->from($twitterUserIds);
        }

        $this->onTweet = $whenTweets;

        return $this;
    }
```

**Reply by numairawan** [reply]
It works for me.
